### PR TITLE
[BEAM-13886] unit tests for trigger package

### PR DESCRIPTION
**Pull request by riteshghorse**
Unit tests for trigger package. Part of effort to improve code coverage of Go SDK.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by riteshghorse** [comment]
R: @jrmccluskey @damccorm

**Review by jrmccluskey — APPROVED** [review]
LGTM!

File: sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+func TestAfterCount(t *testing.T) {
+	tr := AfterCount(1)
+	want := int32(1)
+	if got := tr.ElementCount(); got != want {
+		t.Errorf("element count not configured correctly. got %v, want %v", got, want)
+	}
+}
+
+func TestAfterProcessingTime(t *testing.T) {
```
Comment (damccorm) [inline sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go:30]: I think this is actually more testing the full behavior of AfterProcessingTimeTrigger, so I'd probably update to `TestAfterProcessingTimeTrigger`

File: sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	tr := AfterProcessingTime()
+	tr.PlusDelay(time.Millisecond).AlignedTo(time.Millisecond, time.Now())
+	want := 2
+	if got := len(tr.TimestampTransforms()); got != want {
+		t.Errorf("timestamp transforms not configured correctly. got %v, want %v", got, want)
+	}
```
Comment (damccorm) [inline sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go:36]: Could we verify that the transforms are actually correct as well? e.g. that the first one is a DelayTransform with delay of 1 and the second one is an AlignToTransform with the correct values as well

File: sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+func TestRepeat(t *testing.T) {
+	subTr := AfterCount(1)
+	tr := Repeat(subTr)
+
+	if got := tr.SubTrigger(); got != subTr {
+		t.Errorf("subtrigger not configured correctly. got %v, want %v", got, subTr)
+	}
+}
+
+func TestAfterEndOfWindow(t *testing.T) {
```
Comment (damccorm) [inline sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go:48]: Same comment about naming - could we name this TestAfterEndOfWindowTrigger?

Comment (damccorm) [reply sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go:48]: Normally I'd push to separate testing different functions in different tests, but things are so simple here that I don't think we need that

File: sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go (hunk 2 above)
Comment (riteshghorse) [reply sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go:36]: Yes.. good suggestion

File: sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go (hunk 3 above)
Comment (lostluck) [reply sdks/go/pkg/beam/core/graph/window/trigger/trigger_test.go:48]: And ultimately, we can have sub tests, if we want grouping  t.Run can be executed just fine outside of a loop, and create arbitrary related sub tests.

**Comment by riteshghorse** [comment]
PTAL

**Review by damccorm — APPROVED** [review]
Thanks for the updates! LGTM

**Review by lostluck — APPROVED** [review]

**Pull request merged**